app/services/product_filter_service.rb

```diff
@@ -0,0 +1,41 @@
+class ProductFilterService
+  def initialize(params)
+    @params = params
+  end
+
+  def filter
+    products = Product.all
+    
+    if @params[:category].present? && @params[:category] != 'Category'
+      products = products.where(category: @params[:category])
+    end
+    
+    if @params[:vegetarian].present? && @params[:vegetarian] != 'Vegetarian'
```

Comment:
Are these url params? If yes, what would happen if someone writes the url as `?vegetarian=testingabug`. I think it will enter the if because `@params[:vegetarian].present? && @params[:vegetarian] != 'Vegetarian'` will be true and it will end up returning all non-vegetarian products as `@params[:vegetarian] == 'true'` will be false.

Reply:
Also, is there any case specifically to check for `@params[:vegetarian] != 'Vegetarian'`? Is there a chance you'll be sending the url as `?vegetarian=Vegetarian`? I think from the frontend you should try to send the parameter as only true or false 🤔 and do a check for that